### DISCUSSION TITLE
change default punctuation color?

### DISCUSSION THREAD

**Discussion by 1sixth** [eza-community/eza#209] [Development]
eza uses `Black.bold()` to display punctuations, making them unreadable in some cases:

https://github.com/eza-community/eza/blob/724b6ddbf572425ac35ebe8969839566a9a1de98/src/theme/default_theme.rs#L80

Tested on alacritty with default config:

![1](https://github.com/eza-community/eza/assets/67363572/4610d711-f478-42d1-ad3a-42f877db7f26)

Tested on kitty with default config:

![2](https://github.com/eza-community/eza/assets/67363572/b8072db1-0657-4568-9df6-29d53a1517b2)

Tested on xterm with default config:

![3](https://github.com/eza-community/eza/assets/67363572/4548ebdc-af99-4023-91e1-2941915dd7c6)

Relevant information:

- eza: 0.11.0
- alacritty: 0.12.2
- kitty: 0.29.1
- xterm: 384

**Comment by cafkafk** [comment]
This isn't as much a bug as it's a question of whether or not the color should be changed, and thus discussions are perhaps better suited to gather community input.

**Comment by smackysnacks** [comment]
The fix for this is to set `EXA_COLORS="xx=''"`, though I think `eza` should revert back to the default `exa` behavior. The `eza` change is broken by default for too many setups, including my own.

**Comment by cafkafk** [comment]
This does keep coming up so perhaps we should change it. Seen no one that are for keeping this.

**Comment by cafkafk** [comment]
It is clear now from community feedback that the punctuation colors need to be changed, and so I'm closing this for now. Please help contributors by providing feedback in #240.

**Comment by z0vsky** [comment]
I was about to create a new issue and found this discussion. Bumping it, and adding my findings:

### Context 

Tested on:
- Gnome Terminal (Ubuntu 24.04) with the built-in Solarized-dark color scheme and Solarized ANSI palette.
- iTerm2 (macOS 14.6) with the built-in Solarized-dark color setup.

### Expected results

Other tools like `tree`, `exa --tree`, and `lsd --tree` correctly use the **foreground** color for tree branch lines, which makes them visible. The same goes with the '-' char from the file permission.

#### Legacy `tree` binary
<img width="432" alt="tree_solarized-dark" src="https://github.com/user-attachments/assets/80532fcc-d81e-4562-b936-3df09caae0b3">

#### legacy `exa --tree`
![exa_solarized-dark](https://github.com/user-attachments/assets/605ca70f-b58b-4425-8e5c-e00b32597194)

#### LS Deluxe

`lsd --tree`:
<img width="443" alt="lsd-tree_solarized-dark" src="https://github.com/user-attachments/assets/2c56b6c1-fd03-4b5e-8353-f324bd20fab0">

`lsd --long`:
<img width="852" alt="lsd-long_solarized-dark" src="https://github.com/user-attachments/assets/ac1e1c14-340a-48ea-adc3-d46330289444">

### Actual results

In `eza --tree`, the branch lines are rendered using the "black bright" ANSI color from the terminal's color scheme.
On dark colors schemes and palettes, this messes with the output as foreground blends with (or is the same as) the background color.

<img width="401" alt="exa-tree_solarized-dark" src="https://github.com/user-attachments/assets/4da8cd53-20d9-48d8-b96b-94c8bf930985">

Note: it also happens with `chmod` permission characters when permission is `0` (thus represented by a '-' character).
<img width="659" alt="eza-long_solarized-dark" src="https://github.com/user-attachments/assets/576ca559-e1ed-482e-9036-37569f591d69">

### Attempts

I tried without and with [dircolors-solarized](https://github.com/seebi/dircolors-solarized), but this had no effect on the tree's color rendering.

#### Temporary workaround

Set Black Bright to the same color as Foreground, but this could mess with other tools' color output rendering and readability.

<img width="1023" alt="temporary-workaround" src="https://github.com/user-attachments/assets/0764be16-fcfa-4160-88c4-20652d4f3ec0">

### Suggested workaround

`eza` should set the color to Foreground as it should be expected. Should someone use something else, there a free to do so with `theme.yml` or `EXA_COLORS` as [suggested](https://github.com/orgs/eza-community/discussions/209#discussioncomment-6938006) by @smackysnacks.
